maniphttp/manipulator.go

```diff
@@ -667,15 +667,19 @@ func (s *httpManipulator) send(
 				goto RETRY
 
 			case io.ErrUnexpectedEOF, io.EOF:
-				lastError = manipulate.NewErrCannotCommunicate(snip.Snip(err, s.currentPassword()).Error())
+				if lastError == nil {
+					lastError = manipulate.NewErrCannotCommunicate(snip.Snip(err, s.currentPassword()).Error())
+				}
 				goto RETRY
 			}
 
 			// We check for error types.
 			switch uerr.Err.(type) {
 
 			case net.Error:
-				lastError = manipulate.NewErrCannotCommunicate(snip.Snip(err, s.currentPassword()).Error())
+				if lastError == nil {
+					lastError = manipulate.NewErrCannotCommunicate(snip.Snip(err, s.currentPassword()).Error())
+				}
```

Comment:
@primalmotion 👀 

Not sure why `lastError` was being overwritten here, but this change makes the unit tests 🍏 and consistent.

Previously, this call flow would cause tests (in this example, a test expecting to get back a `manipulate.ErrTooManyRequests` error) to fail:

Flow:
- Hit server, got back 429 (ok I will retry)
- Hit server, got back 429 (ok I will retry)
- Hit server, get back 429 (ok I will retry)
- Hit server, get back 429 (ok I will retry)
- (last retry attempt) This guy now fails randomly due to a network error or the sub-context deadline exceeding - the lastError is now overwritten w/ `manipulate.NewErrCannotCommunicate`